pkg/registry/falcon_registry/registry.go

```diff
@@ -117,6 +121,57 @@ func guessLastTag(tags []string, filter func(string) bool) (string, error) {
 func listDockerTags(ctx context.Context, sys *types.SystemContext, imgRef types.ImageReference) ([]string, error) {
 	tags, err := docker.GetRepositoryTags(ctx, sys, imgRef)
 	if err != nil {
+		// Artifactory incorrectly adds the digest to the tag list, so we need to handle this case
+		// by fetching the tags from the registry directly
+		// This is a workaround, and should be removed once Artifactory is fixed
+		if len(tags) == 0 {
+			var jsonRes map[string]interface{}
+			reg, _, _ := strings.Cut(imgRef.StringWithinTransport(), ":")
+			reg = strings.Replace(reg, "crowdstrike.com/", "crowdstrike.com/v2/", 1)
+
+			tr := &http.Transport{
+				TLSClientConfig: &tls.Config{
+					MinVersion: tls.VersionTLS12,
+				},
+			}
+
+			client := &http.Client{Transport: tr}
+			req, err := http.NewRequest("GET", fmt.Sprintf("https:%s/tags/list", reg), nil)
+			if err != nil {
+				return nil, err
+			}
+
+			req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", sys.DockerAuthConfig.Password))
+			resp, err := client.Do(req)
+			if err != nil {
+				return nil, fmt.Errorf("Unable to get http response to list container registry tags: %v", err)
+			}
+			defer resp.Body.Close()
+
+			bodyText, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return nil, fmt.Errorf("Unable to read response body to list registry tags: %v", err)
+			}
+
+			if err := json.Unmarshal(bodyText, &jsonRes); err != nil {
```

Comment:
Is the response from our API always in `JSON` format? 
(i,e) do we need to pass format in our request?

Reply:
Yes. A docker registry check is always in json... added it to the header.